### ISSUE TITLE
How to persist recursive entities

### ISSUE THREAD

**Issue by lev-kuznetsov** [typeorm/typeorm#1167]
I have two entities related by a one-to-many. When I try to persist the 'one-side' with an array filled in of the 'many-side' I only see the 'one-side' persisted. Here are my entities:

```
@ClassEntityChild() // primary key in Item
@DiscriminatorValue('dataset')
export class Dataset extends Item {
  @Column(type => Dimension)
  columns: Dimension

  @Column(type => Dimension)
  rows: Dimension

  @OneToMany(type => Block, block => block.dataset)
  blocks: Block[]
}

@Entity({
  orderBy: {order:'ASC'}
})
export class Block {
  @PrimaryGeneratedColumn()
  private id: number

  @Column()
  order: number

  @ManyToOne(type => Dataset, dataset => dataset.blocks, {cascadeAll: true})
  dataset: Dataset

  @Column('simple-array')
  values: number[]
}

@Entity()
export class Dimension {
  @Column('simple-array')
  keys: string[]
}
```

And this is the code:

```
      let datasets = connection.getRepository(Dataset)
      let dataset = new Dataset()
      dataset.name = 'bar'
      let columns = dataset.columns = new Dimension()
      columns.keys = ['c1', 'c2', 'c3']
      let rows = dataset.rows = new Dimension()
      rows.keys =['r1', 'r2', 'r3', 'r4']
      let block = (dataset.blocks = [new Block()])[0]
      block.order = 0
      block.values = [1.1, 1.2, 1.3, 2.1, 2.2, 2.3, 3.1, 3.2, 3.3, 4.1, 4.2, 4.3]
      block.dataset = dataset
      try {
        console.log(dataset)
        await datasets.persist(dataset)
        console.log('saved!')
        let ds = await datasets.find()
        ds.map(d => console.log(d)
      } catch (err) {
        console.log(err)
      }
```

Looking at the logs and the database, the block isn't inserted. From looking at examples it's my understanding that this is supposed to work; what am I doing wrong? I also tried using the repository.save() method I'm using sqlite, logs are:

```
{
  "name": "bar",
  "columns": {"keys": ["c1",  "c2",  "c3"]},
  "rows": {"keys": ["r1", "r2", "r3", "r4"]},
  "blocks": [
    {
      "order": 0,
      "values": [1.1, 1.2, 1.3, 2.1, 2.2, 2.3, 3.1, 3.2, 3.3, 4.1, 4.2, 4.3]
    }
  ]
}
executing query:  BEGIN TRANSACTION
executing query:  INSERT INTO "item"("name", "type") VALUES ($1,$2) -- PARAMETERS: ["bar","dataset"]
executing query:  INSERT INTO "dataset"("columnsKeys", "rowsKeys", "itemId") VALUES ($1,$2,$3) -- PARAMETERS: ["c1,c2,c3","r1,r2,r3,r4",1]
executing query:  COMMIT
saved!
```

**Comment by pleerock** [comment]
class table inheritance is one of the most problematic functionality we have currently. We are planning to investigate and/or fix issues in 0.4.0

**Comment by pleerock** [comment]
Class table inheritance support will be completely dropped in 0.2.0 due to complexity and value it brings. There are no plans to add CTI support in the near future as well. You can still use single table inheritance, concrete table inheritance or inheritance via embeddeds
